Assignment2/src/main/java/bgu/spl/mics/application/objects/GPU.java

```diff
@@ -21,84 +24,131 @@
     private Type type;
     public Model model;
     Cluster cluster;
+    private GPUTimeService gpuTimeService;
 
     // Memory:
     int vramCapacity;
-    Collection<DataBatch> disk;
-    Collection<DataBatch> trainedDisk;
-    Collection<DataBatch> vRam;
+    LinkedBlockingQueue<DataBatch> disk;
+    LinkedBlockingQueue<DataBatch> trainedDisk;
+    LinkedBlockingQueue<DataBatch> vRam;
 
 
     @Override
     public void insertModel(Model model) {
-
+        if (disk.size() == 0 && model.status != Model.statusEnum.Training){
+            this.model = model;
+        }
     }
 
     @Override
     public void splitToBatches(Data data) {
-
+        int numberOfBatches = data.getSize()/1000;
+        for (int i = 0; i<numberOfBatches;i++){
+            //TODO consult with Noyhoz

```

Comment:
we have wto creae data batches with index of the first sample (Example:
DataBatch :
data =data
startingIndex = 0

DataBatch :
data =data
startingIndex = 1000

)